tests/Auth0/Listeners/CreateClientsTest.php

```diff
@@ -67,27 +67,44 @@ public function it_creates_a_new_client_in_every_configured_tenant(): void
+
         $this->httpClient->expects($this->exactly(6))
             ->method('sendRequest')
             ->willReturnOnConsecutiveCalls(
-                new Response(201, [], Json::encode(['client_id' => 'client-id-1', 'client_secret' => 'client-secret-1'])),
+                new Response(201, [], Json::encode(['id' => $clientIds[0], 'client_id' => 'client-id-1', 'client_secret' => 'client-secret-1'])),
                 new Response(201, [], Json::encode(['id' => 'grant-id-1'])),
-                new Response(201, [], Json::encode(['client_id' => 'client-id-2', 'client_secret' => 'client-secret-2'])),
+                new Response(201, [], Json::encode(['id' => $clientIds[1], 'client_id' => 'client-id-2', 'client_secret' => 'client-secret-2'])),
                 new Response(201, [], Json::encode(['id' => 'grant-id-2'])),
-                new Response(201, [], Json::encode(['client_id' => 'client-id-3', 'client_secret' => 'client-secret-3'])),
+                new Response(201, [], Json::encode(['id' => $clientIds[2], 'client_id' => 'client-id-3', 'client_secret' => 'client-secret-3'])),
                 new Response(201, [], Json::encode(['id' => 'grant-id-3'])),
             );
 
+
         $expectedClients = [
-            new Auth0Client($integrationId, 'client-id-1', 'client-secret-1', Auth0Tenant::Acceptance),
-            new Auth0Client($integrationId, 'client-id-2', 'client-secret-2', Auth0Tenant::Testing),
-            new Auth0Client($integrationId, 'client-id-3', 'client-secret-3', Auth0Tenant::Production),
+            new Auth0Client($clientIds[0], $integrationId, 'client-id-1', 'client-secret-1', Auth0Tenant::Acceptance),
+            new Auth0Client($clientIds[1], $integrationId, 'client-id-2', 'client-secret-2', Auth0Tenant::Testing),
+            new Auth0Client($clientIds[2], $integrationId, 'client-id-3', 'client-secret-3', Auth0Tenant::Production),
         ];
 
         $this->clientRepository->expects($this->once())
             ->method('save')
-            ->with(...$expectedClients);
+            ->with(
+                $this->callback(fn (Auth0Client $client) => $this->compareAuth0Client($client, $expectedClients[0])),
+                $this->callback(fn (Auth0Client $client) => $this->compareAuth0Client($client, $expectedClients[1])),
+                $this->callback(fn (Auth0Client $client) => $this->compareAuth0Client($client, $expectedClients[2])),
+            );
 
         $this->createClients->handle(new IntegrationCreated($integrationId));
     }
+
+    private function compareAuth0Client(Auth0Client $client, Auth0Client $expectedClient): true
+    {
+        $this->assertEquals($client->clientId, $expectedClient->clientId);
+        $this->assertEquals($client->integrationId, $expectedClient->integrationId);
+        $this->assertEquals($client->tenant, $expectedClient->tenant);
+        $this->assertEquals($client->clientSecret, $expectedClient->clientSecret);
+
+        return true;
```

Comment:
No need for a return value.

Reply:
$this->callback() requires a boolean, so I thought it was more elegant doing it this way.
Otherwise I cannot use a single line callback fn.

Reply:
Ok! I see now that the closure is typed with a boolean return value.